core/sql/generator/GenRelScan.cpp

```diff
@@ -214,22 +215,32 @@ int HbaseAccess::createAsciiColAndCastExpr(Generator * generator,
       ((CharType*)newGivenType)->setUpshifted(FALSE);
     }
 
+  if (newGivenType->getTypeQualifier() == NA_CHARACTER_TYPE &&
+      CmpCommon::getDefaultString(HIVE_FILE_CHARSET) == "GBK")
```

Comment:
It might be good to add a check here that HIVE_FILE_CHARSET must be an empty string or equal to HIVE_DEFAULT_CHARSET or GBK. In all other cases we would probably not do what the user expected.